internal/helm/repository/client_opts.go

```diff
@@ -0,0 +1,124 @@
+package repository
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+
+	"github.com/fluxcd/pkg/oci"
+	"github.com/google/go-containerregistry/pkg/authn"
+	helmgetter "helm.sh/helm/v3/pkg/getter"
+	helmreg "helm.sh/helm/v3/pkg/registry"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	helmv1 "github.com/fluxcd/source-controller/api/v1beta2"
+	"github.com/fluxcd/source-controller/internal/helm/getter"
+	"github.com/fluxcd/source-controller/internal/helm/registry"
+)
+
+// HelmClientOpts contains the various options to use while constructing
+// a Helm client.
+type HelmClientOpts struct {
+	Authenticator          authn.Authenticator
+	Keychain               authn.Keychain
+	LoginOpt               helmreg.LoginOption
+	TlsConfig              *tls.Config
+	GetterOpts             []helmgetter.Option
+	EmitDeprecationWarning bool
```

Comment:
can't decide whether this should be here or `GetHelmClientOpts()` should return it as a third value

Reply:
Or as an error type?

Reply:
i feel like that would introduce unnecessary complexity, especially for something which will be removed in some time anyway. i'm happy to keep it as a part of the options, if you are.

Reply:
It would allow you to specifically construct the warning within the context of the actual code, instead of making an assumption in the controller code and thereby breaking the separation of concerns.